### PR TITLE
feat: add counter trigger for rav request

### PR DESCRIPTION
**Pull request by carlosvdr**

### CODE REVIEW

File: tap-agent/src/agent/sender_fee_tracker.rs [hunk 1]
```diff
@@ -92,8 +109,8 @@ impl SenderFeeTracker {
     pub fn update(&mut self, id: Address, fee: u128) {
         if fee > 0 {
             // insert or update, if update remove old fee from total
-            if let Some(old_fee) = self.id_to_fee.insert(id, fee) {
-                self.total_fee -= old_fee;
+            if let Some(old_fee) = self.id_to_fee.insert(id, FeeCounter { fee, count: 0 }) {
+                self.total_fee -= old_fee.fee;
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_fee_tracker.rs:113]: It needs to have an update here for `pub fn update(&mut self, id: Address, fee: u128, counter: usize)`. SQL query and UnaggregatedReceipts need to be updated as well.

**Review by gusinacio — CHANGES_REQUESTED** [review]

File: tap-agent/src/agent/sender_allocation.rs [hunk 2]
```diff
@@ -417,6 +421,7 @@ impl SenderAllocationState {
                 .unwrap_or(BigDecimal::from(0))
                 .to_string()
                 .parse::<u128>()?,
+            counter: res.count.unwrap_or(0).to_string().parse::<usize>()?,
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_allocation.rs:424]: why do you need to_string().parse::<usize>()?

File: tap-agent/src/agent/sender_allocation.rs [hunk 3]
```diff
@@ -398,6 +400,7 @@ impl SenderAllocationState {
...
```
Comment (gusinacio) [inline tap-agent/src/agent/sender_allocation.rs:403]: same here, why not just convert to usize?

File: tap-agent/src/agent/sender_allocation.rs (hunk 2 above)
Comment (carlosvdr) [reply tap-agent/src/agent/sender_allocation.rs:442]: tried doing just res.count.unwrap_or(0 as usize) but it would still be u64, got any suggestion?

**Review by gusinacio — COMMENTED** [review]
I'd like to have metrics regarding the amount of receipts per allocation. This can be done on a following PR.

**Review by gusinacio — APPROVED** [review]
LGTM

**Pull request merged**